testing/src/test/java/io/stargate/it/cql/SimpleStatementTest.java

```diff
@@ -0,0 +1,172 @@
+    for (int i = 0; i < 100; i++) {
+      session.execute("INSERT INTO test (k, v) VALUES (?, ?)", KEY, i);
+    }
+
+    // table with simple primary key, single cell.
+    session.execute("CREATE TABLE IF NOT EXISTS test2 (k text primary key, v int)");
+  }
+
+  @Test
+  public void should_use_positional_values() {
+    SimpleStatement statement = SimpleStatement.newInstance("SELECT v FROM test WHERE k=?", KEY);
+    ResultSet resultSet = session.execute(statement);
+    assertThat(resultSet).hasSize(100);
+  }
+
+  @Test
+  public void should_allow_nulls_in_positional_values() {
+    session.execute("INSERT into test2 (k, v) values (?, ?)", KEY, null);
+    Row row = session.execute("select k,v from test2 where k=?", KEY).one();
+    assertThat(row).isNotNull();
+    assertThat(row.isNull("v")).isTrue();
+  }
+
+  @Test
+  @Disabled(
+      "C* 3.11 throws 'InvalidQueryException: Invalid amount of bind variables', Stargate doesn't throw")
+  public void should_fail_when_too_many_positional_values_provided() {
+    assertThatThrownBy(
+            () -> session.execute("INSERT into test2 (k, v) values (?, ?)", KEY, 1, 2, 3))
+        .isInstanceOf(InvalidQueryException.class);
+  }
+
+  @Test
+  @Disabled(
+      "C* 3.11 returns an InvalidQueryException, Stargate returns a ServerError(IndexOutOfBoundsException)")
+  public void should_fail_when_not_enough_positional_values_provided() {
+    // For SELECT queries, all values must be filled
+    assertThatThrownBy(() -> session.execute("SELECT * from test where k = ? and v = ?", KEY))
+        .isInstanceOf(InvalidQueryException.class);
+  }
+
+  @Test
+  public void should_use_named_values() {
+    SimpleStatement statement =
+        SimpleStatement.newInstance("SELECT v FROM test WHERE k=:k", ImmutableMap.of("k", KEY));
+    ResultSet resultSet = session.execute(statement);
+    assertThat(resultSet).hasSize(100);
+  }
+
+  @Test
+  public void should_allow_nulls_in_named_values() {
+    session.execute(
+        SimpleStatement.builder("INSERT into test2 (k, v) values (:k, :v)")
+            .addNamedValue("k", KEY)
+            .addNamedValue("v", null)
+            .build());
+
+    Row row = session.execute("select k,v from test2 where k=?", KEY).one();
+    assertThat(row).isNotNull();
+    assertThat(row.isNull("v")).isTrue();
+  }
+
+  @Test
+  @Disabled(
+      "C* 3.11 returns an InvalidQueryException, Stargate returns a ServerError(IndexOutOfBoundsException)")
+  public void should_fail_when_named_value_missing() {
+    // For SELECT queries, all values must be filled
+    assertThatThrownBy(
+            () ->
+                session.execute(
+                    SimpleStatement.newInstance(
+                        "SELECT * from test where k = :k and v = :v", ImmutableMap.of("k", KEY))))
+        .isInstanceOf(InvalidQueryException.class);
+  }
+
+  @Test
+  public void should_use_paging_state() {
```

Comment:
Codacy found an issue: [The JUnit 5 test method name 'should_use_paging_state' doesn't match '[a-z][a-zA-Z0-9]*'](https://app.codacy.com/gh/stargate/stargate/pullRequest?prid=6263918)